mininet/net.py

```diff
@@ -729,9 +729,10 @@ def wifiDirect(self, sta, cls=None, **params):
             \ndevice_name=%s\
             \ndevice_type=1-0050F204-1\
             \np2p_no_group_iface=1' % (sta)
-        apcommand = cmd + ("\' > %s_wifiDirect.conf" % sta)
+        confname = "mn%d_%s_wifiDirect.conf" % (os.getpid(), sta)
+        apcommand = cmd + ("\' > %s" % confname)
```

Comment:
I just moved the config name to an extra variable, apcommand is your variable name.